private_gpt/server/chat/chat_service.py

```diff
@@ -58,63 +57,69 @@ def __init__(
         )
 
     def _chat_engine(
-        self, context_filter: ContextFilter | None = None
-    ) -> BaseChatEngine:
-        vector_index_retriever = self.vector_store_component.get_retriever(
-            index=self.index, context_filter=context_filter
-        )
-        return ContextChatEngine.from_defaults(
-            retriever=vector_index_retriever,
-            service_context=self.service_context,
-            node_postprocessors=[
-                MetadataReplacementPostProcessor(target_metadata_key="window"),
-            ],
-        )
-
-    def stream_chat(
         self,
-        messages: list[ChatMessage],
+        system_prompt: str | None = None,
         use_context: bool = False,
         context_filter: ContextFilter | None = None,
-    ) -> CompletionGen:
+    ) -> BaseChatEngine:
         if use_context:
-            last_message = messages[-1].content
-            chat_engine = self._chat_engine(context_filter=context_filter)
-            streaming_response = chat_engine.stream_chat(
-                message=last_message if last_message is not None else "",
-                chat_history=messages[:-1],
+            vector_index_retriever = self.vector_store_component.get_retriever(
+                index=self.index, context_filter=context_filter
             )
-            sources = [
-                Chunk.from_node(node) for node in streaming_response.source_nodes
-            ]
-            completion_gen = CompletionGen(
-                response=streaming_response.response_gen, sources=sources
+            return ContextChatEngine.from_defaults(
+                system_prompt=system_prompt,
```

Comment:
`system_prompt` here is really a `default_system_prompt`

Reply:
What do you mean? It is not a default system prompt, it is the one passed by the user.